addons/web/static/lib/ace/mode-sql.js

```diff
@@ -0,0 +1,95 @@
+define("ace/mode/sql_highlight_rules",["require","exports","module","ace/lib/oop","ace/lib/lang","ace/mode/text_highlight_rules"], function(require, exports, module) {
+    "use strict";
+
+    var oop = require("../lib/oop");
+    var TextHighlightRules = require("./text_highlight_rules").TextHighlightRules;
+
+    var SqlHighlightRules = function() {
+
+        var keywords = (
+            "select|insert|update|delete|from|where|and|or|group|by|order|limit|offset|having|as|case|" +
+            "when|then|else|end|type|left|right|join|on|outer|desc|asc|union|create|table|primary|key|if|" +
+            "foreign|not|references|default|null|inner|cross|natural|database|drop|grant|distinct"
+        );
+
+        var builtinConstants = (
+            "true|false"
+        );
+
+        var builtinFunctions = (
+            "avg|count|first|last|max|min|sum|ucase|lcase|mid|len|round|rank|now|format|" +
+            "coalesce|ifnull|isnull|nvl"
+        );
+
+        var dataTypes = (
+            "int|numeric|decimal|date|varchar|char|bigint|float|double|bit|binary|text|set|timestamp|" +
+            "money|real|number|integer"
+        );
+
+        var keywordMapper = this.createKeywordMapper({
+            "support.function": builtinFunctions,
+            "keyword": keywords,
+            "constant.language": builtinConstants,
+            "storage.type": dataTypes
+        }, "identifier", true);
```

Comment:
If it's not gonna change, we should use `const` unless there is something I don't know about that is impacted.

Reply:
This file is mostly a copy paste from ACE, except some minor changes to integrate it in Odoo (such as the use off `require` etc)